### PR TITLE
Update async slicing discussion section with PR feedback

### PR DESCRIPTION
**Pull request by andy-sweet**
# Description

This updates the async slicing NAP's discussion section with a summary of the feedback in https://github.com/napari/napari/pull/4892.

It also identifies two critical/blocking discussion points that need to be resolved before the NAP can be approved.

1. Should `Dims.current_step` represent the last request or last response?
2. Can we easily support users that may depend on synchronous slicing (e.g. napari-animation plugin)?

These are somewhat related to breaking existing behavior.

Other discussion points are deemed important enough to be captured, but not blocking.

I also added an alternative idea to capture a previous open question that [got some comments](https://github.com/napari/napari/pull/4892/files#r935338577) and closed the other open questions, as they are now captured in the discussion.

# References

Part of #4795

### CODE REVIEW

File: docs/naps/4-async-slicing.md
Comment (andy-sweet) [inline docs/naps/4-async-slicing.md:20]: Missed in last PR from comment: https://github.com/napari/napari/pull/4892/files#r935764844

**Comment by andy-sweet** [comment]
I assigned those whose comments I tried to summarize. I suspect we only need one or two approvals for this to get merged.

**Review by sofroniewn — APPROVED** [review]
LGTM

**Review by Carreau — APPROVED** [review]

**Comment by andy-sweet** [comment]
As this has two approvals, I'll merge this in 24 hours unless I hear otherwise.

File: docs/naps/4-async-slicing.md
Comment (Czaki) [inline docs/naps/4-async-slicing.md:596]: One small remark. This introduces Qt dependency outside `_qt` directory. I understand that it could be a generalization. But if we decide to introduce Qt to vispy module, then we could just make `VispyImageLayer` A Qt Object and direct use signals that have better performance and may simplify the problem of synchronized mode.

**Review by Czaki — APPROVED** [review]
just one comment

File: docs/naps/4-async-slicing.md
Comment (andy-sweet) [reply docs/naps/4-async-slicing.md:596]: Good point. I added a new paragraph in this section to capture that idea.

**Pull request merged**
